Added support for `coc.nvim` or other LSP providers

lmburns opened the pull request:
If a user chooses not to use `lspconfig`, then this plugin is unusable. I added a check to make sure that `lspconfig` is installed before it is required. This way people are able to configure something like `coc.nvim` to support this plugin.

I use `packer` and am unsure how to check whether or not `lspconfig` is supported with `paq`. So I just checked whether or not it had been picked up by Lua itself instead.

kylo252 commented on lua/lua-dev/sumneko.lua:
```diff
@@ -63,8 +63,10 @@ function M.types()
 function M.setup(opts)
-  local lspconfig = require("lspconfig")
-  return {
+  local on_new_config
+
+  if (_G.packer_plugins and _G.packer_plugins["lspconfig"]) or package.loaded["lspconfig"] then
+    local lspconfig = require("lspconfig")
```
you can use `status_ok` if you want to have custom handling, or replace it with an underscore `_` otherwise

lmburns commented:
I fixed it, having a much simpler approach. I'm also unsure how formatting issues came about. I've fixed this before, but for some reason can't get it to work now, so that way only the lines I directly modified will be showed in the blame

max397574 commented:
lgtm
